### PR TITLE
Improving class structure, team functionality and plots, fixing language handling

### PR DESCRIPTION
**Pull request by maciekbanas**
Basically changes from this PRs:
#212 - improving plots.
#214 - deprecate Engine class and pull repos by team directly from GraphQL (before it was filtering ex-post)
#216 - fixing filtering by language and pulling repos by team from GraphQL.
and some small fixes.

### CODE REVIEW

File: devel/example_workflow.R [hunk 1]
```diff
@@ -47,11 +53,8 @@ setup(git_stats,
 get_repos(git_stats)
 
 # you can plot repos sorted by last activity
-
 plot_repos(git_stats)
```
Comment (marcinkowskak) [inline devel/example_workflow.R:56]: There's something wrong with this function as the plot I get looks like that: 
image
 but when you compare it with the code that was on this #212 PR you may see the difference in the code. Like all the changes made for this function didn't get into devel branch at all.

File: R/EngineGraphQLGitHub.R [hunk 2]
```diff
@@ -116,20 +116,32 @@ EngineGraphQLGitHub <- R6::R6Class("EngineGraphQLGitHub",
     }
   ),
   private = list(
+    # @field token A token authorizing access to API.
```
Comment (marcinkowskak) [inline R/EngineGraphQLGitHub.R:119]: Should it be commented or it was to be a documentation? If the latter then **'** character is missing.

Comment (maciekbanas) [reply R/EngineGraphQLGitHub.R:119]: Roxygen does not support well documenting R6 private methods - it pops out a lot of warnings when you `devtools::document()` - there was an issue created for that on `roxygen` github site but with no reasonable workout (just closed, one of the solutions proposed is to make documentation as simple comments - I think the way it is done in Python) https://github.com/r-lib/roxygen2/issues/1016

File: devel/example_workflow.R (hunk 1 above)
Comment (maciekbanas) [reply devel/example_workflow.R:56]: I handled it - the root cause was not in `plot_repos` but in pulling repos by team which was recently refactored with pulling repos by user directly via GraphQL in #214 - in some cases the repos were pulled multiple times (for many user same repos) - and that distorted the plot. The solution was simply applying distinct rows to output repos table.

Comment (marcinkowskak) [reply devel/example_workflow.R:56]: Oh, that would explain double rows, indeed. Looks good now! :)

**Review by marcinkowskak — APPROVED** [review]
Looks good to me now, data is pulled nicely and new way of showing progress bar looks awesome! :)

**Pull request merged**